x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/reducer.ts

```diff
@@ -538,17 +547,53 @@ export function reducer(state: State, action: Action): State {
 
     case ACTION.SWITCH_TO_ADVANCED_EDITOR:
       let { jobConfig } = state;
-      const isJobConfigEmpty = isEmpty(state.jobConfig);
-      if (isJobConfigEmpty) {
-        jobConfig = getJobConfigFromFormState(state.form);
-      }
+      jobConfig = getJobConfigFromFormState(state.form);
+      const shouldDisableSwitchToForm = isAdvancedConfig(jobConfig);
+
       return validateAdvancedEditor({
         ...state,
         advancedEditorRawString: JSON.stringify(jobConfig, null, 2),
         isAdvancedEditorEnabled: true,
+        disableSwitchToForm: shouldDisableSwitchToForm,
+        hasSwitchedToEditor: true,
         jobConfig,
       });
 
+    case ACTION.SWITCH_TO_FORM:
+      const { jobConfig: config, jobIds } = state;
+      const { jobId } = state.form;
+      // @ts-ignore
+      const formState = getFormStateFromJobConfig(config, false);
+
+      if (typeof jobId === 'string' && jobId.trim() !== '') {
+        formState.jobId = jobId;
+      }
+
+      formState.jobIdExists = jobIds.some((id) => formState.jobId === id);
+      formState.jobIdEmpty = jobId === '';
+      formState.jobIdValid = isJobIdValid(jobId);
+      formState.jobIdInvalidMaxLength = !!maxLengthValidator(JOB_ID_MAX_LENGTH)(jobId);
+
+      formState.destinationIndexNameEmpty = formState.destinationIndex === '';
+      formState.destinationIndexNameValid = isValidIndexName(formState.destinationIndex || '');
+      formState.destinationIndexPatternTitleExists =
+        state.indexPatternsMap[formState.destinationIndex || ''] !== undefined;
+
+      if (formState.numTopFeatureImportanceValues !== undefined) {
+        formState.numTopFeatureImportanceValuesValid = validateNumTopFeatureImportanceValues(
+          formState.numTopFeatureImportanceValues
+        );
+      }
+
+      return validateForm({
+        ...state,
+        // @ts-ignore
+        form: formState,
```

Comment:
Can you explain and add a comment why the ignore is necessary? Do you plan to solve it in a follow up?

Reply:
Same as above.